File: RoundTable SDK for DotNET/src/interfaces/IScheduleManager.cs
using System;

namespace RoundTable.ProjectServer.Services.ScheduleManager
{
	/// <summary>
	/// Summary description for IScheduleManager.
	/// </summary>
	public interface IScheduleManager
	{
		int addSchedule(Schedule schedule);
   
		void dropSchedule(int scheduleId);
   
		void modifySchedule(Schedule schedule);
   
		Schedule getSchedule(int scheduleId);
   
		Schedule[] getAllSchedulesForProject(int projectId);
   
		//
		// Comments Convenience Methods
		//
		int addCommentForSchedule(Schedule schedule, Comment comment);
   
		Comment[] getCommentsForSchedule(Schedule schedule);
   
		int addCommentForScheduleItem(ScheduleItem scheduleItem, Comment comment);
   
		Comment[] getCommentsForScheduleItem(ScheduleItem scheduleItem);
	}
}

File: RoundTable Project Server/src/com/dommoni/roundtable/projectServer/services/scheduleManager/IScheduleManager.java
package com.dommoni.roundtable.projectServer.services.scheduleManager;

import com.dommoni.roundtable.Comment;
import com.dommoni.roundtable.Document;
import com.dommoni.roundtable.Schedule;
import com.dommoni.roundtable.ScheduleItem;

public interface IScheduleManager
{
   public int addSchedule(Schedule schedule)
   throws Exception;
   
   public void dropSchedule(int scheduleId)
   throws Exception;
   
   public void modifySchedule(Schedule schedule)
   throws Exception;
   
   public Schedule getSchedule(int scheduleId)
   throws Exception;
   
   public Schedule[] getAllSchedulesForProject(int projectId)
   throws Exception;
   
   //
   // Comments Convenience Methods
   //
   public int addCommentForSchedule(Schedule schedule, Comment comment)
   throws Exception;
   
   public Comment[] getCommentsForSchedule(Schedule schedule)
   throws Exception;
   
   public int addCommentForScheduleItem(ScheduleItem scheduleItem, Comment comment)
   throws Exception;
   
   public Comment[] getCommentsForScheduleItem(ScheduleItem scheduleItem)
   throws Exception;
}
/*
 *
 * $Log$
 *
 */